src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -312,17 +317,28 @@ private function getAccessorMethod($class, $property)
     private function getMutatorMethod($class, $property)
     {
         $ucProperty = ucfirst($property);
+        $singulars = $this->getSingulars($ucProperty);
 
         foreach (self::$mutatorPrefixes as $prefix) {
-            try {
-                $reflectionMethod = new \ReflectionMethod($class, $prefix.$ucProperty);
 
-                // Parameter can be optional to allow things like: method(array $foo = null)
-                if ($reflectionMethod->getNumberOfParameters() >= 1) {
-                    return array($reflectionMethod, $prefix);
+            if (null !== $singulars && in_array($prefix, self::$arrayMutatorPrefixes)) {
+                $names = $singulars;
+                $names[] = $ucProperty;
```

Comment:
I prefer to keep it as is: if someone rely on the old behavior (say it has both methods, currently the singular is used), it should not change or it will be a BC break.